### PR TITLE
feat: resolve path function

### LINKED ISSUE #57: Proposal for adding `simplifile.resolve(path)`
I was looking for a way to convert a relative path name to an absolute one and couldn't find it in this library. Erlang has `filename:abspath/1` and Node has `path.resolve()` in `node:path` so it should be possible to add it. It appears they use slightly different semantics for `..` in the final output, but this can be normalized with `filepath.expand`.

### PR DESCRIPTION
**Pull request by folospior**
Created a function - `simplifile.resolve(path)`, which will resolve a relative path to an absolute path through a FFI and expands the path.

Also updated the Gleam version to avoid using the deprecated `gleeunit/should` module. I wanted to implement more tests, but most of them pretty much fail on Windows.

Closes #57

### CODE REVIEW

File: test/simplifile_test.gleam [hunk 1]
```diff
@@ -781,3 +782,9 @@ pub fn unknown_errors_return_unknown_test() {
 @external(erlang, "simplifile_erl", "create_directory")
 @external(javascript, "./simplifile_js.mjs", "createDirectory")
 fn create_directory_with_bad_arg(arg: #(Nil, Nil)) -> Result(Nil, FileError)
+
+pub fn resolve_empty_path_test() {
```
Comment (bcpeinhardt) [inline test/simplifile_test.gleam:785]: I'd definitely like to see some more tests than just an empty path. We don't just need to test that the platform code works, but that the way we're using it, it behaves the same for each target/runtime combo.

File: src/simplifile.gleam [hunk 2]
```diff
@@ -743,3 +743,33 @@ pub fn current_directory() -> Result(String, FileError) {
...
+/// // on Windows
+/// assert resolve("/var/lib/../lucy/./gleam") == Ok("c:/var/lucy/gleam") 
+///
+/// // if the current working directory is `/home/something`
+/// assert resolve("../gleam") == Ok("/home/gleam") 
+///
+/// // Tried to go two directories back, but was only able to go one back. Path is unresolvable.
+/// assert resolve("/tmp/../..") == Error(Nil) 
+/// ```
+pub fn resolve(path path: String) -> Result(String, Nil) {
```
Comment (bcpeinhardt) [inline src/simplifile.gleam]: Functions that return an error in this library generally return the `FileError` type.

File: src/simplifile_js.mjs [hunk 3]
```diff
@@ -398,3 +398,7 @@ function cast_error(error_code) {
       return new $simplifile.Unknown(error_code);
   }
 }
+
+export function resolve(file_path) {
+  return path.resolve(file_path);
```
Comment (bcpeinhardt) [inline src/simplifile_js.mjs]: Some interesting behavior from the node:path docs
> If no path segments are passed, path.resolve() will return the absolute path of the current working directory.

If that's the intended behavior we should test and document it for all targets.

**Review by bcpeinhardt — CHANGES_REQUESTED** [review]

File: src/simplifile_js.mjs (hunk 3 above)
Comment (folospior) [reply src/simplifile_js.mjs]: The way we're using this function doesn't allow the user to not pass a path segment, at most an empty string.

`path.resolve()` works the same as `path.resolve("")` and `filename:absname("")` - all three return the absolute path of the current working directory and there's a test for that.

Comment (folospior) [reply src/simplifile_js.mjs]: I agree it should be documented though

File: test/simplifile_test.gleam (hunk 1 above)
Comment (folospior) [reply test/simplifile_test.gleam:785]: The absolute path of the current working directory has different formats depending on the OS: on windows it'll always start with the drive letter, whereas on Unix it will always start with `/`

The tests would have to be OS-dependent. They do pass on linux though, on both Erlang and JS, so I hope that's enough.

Comment (bcpeinhardt) [reply test/simplifile_test.gleam:785]: Passing on linux and unix is fine for this issue, dealing with windows can be a separate issue I think since we don't have a test setup for that. We should though!

File: src/simplifile_js.mjs (hunk 3 above)
Comment (bcpeinhardt) [reply src/simplifile_js.mjs]: Can you add a jsdoc to this like the other functions have?

File: src/simplifile.gleam [hunk 4]
```diff
@@ -743,3 +743,33 @@ pub fn current_directory() -> Result(String, FileError) {
...
+///
+/// Returns an error if the relative path could not be resolved.
+/// 
+/// # Example:
+/// ```gleam
+/// // Resolving a relative path resolves the full absolute path.
+/// // Assume the current working directory is /home/lucy.
+/// assert resolve("./tmp/../gleam") == Ok("$/home/lucy/gleam")
+///
+/// // On Windows.
```
Comment (bcpeinhardt) [inline src/simplifile.gleam]: Let's not specify the windows behavior until we have testing for windows in place.

**Review by bcpeinhardt — CHANGES_REQUESTED** [review]

File: src/simplifile.gleam (hunk 4 above)
Comment (folospior) [reply src/simplifile.gleam]: Done!

File: src/simplifile_js.mjs (hunk 3 above)
Comment (folospior) [reply src/simplifile_js.mjs]: Done!

**Review by bcpeinhardt — APPROVED** [review]
I can dig it 😎
